src/store/preferences/index.ts

```diff
@@ -53,6 +58,15 @@ export default {
     setNewTags(state: State, tags: string[]) {
       state.taskEditorSectionIdAndTask!.newTags = tags;
     },
+    setDueDate(state: State, date: string) {
```

Comment:
I think we should probably consolidate the date formatting within this mutation, rather than in the other places. For example, if we don't use the DateSelector, we lose out on the formatting that is required by Asana.

Reply:
I added the date formatting as a getter in the setDueDate action instead of the mutation

Reply:
Could you elaborate on why you put it in the action and not in the mutation?